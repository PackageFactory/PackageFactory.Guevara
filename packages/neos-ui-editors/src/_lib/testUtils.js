import {PureComponent, Children} from 'react';
import PropTypes from 'prop-types';
import {SynchronousRegistry, SynchronousMetaRegistry} from '@neos-project/neos-ui-extensibility/src/registry';

const globalRegistry = new SynchronousMetaRegistry();

// i18n Registry
class FakeI18NRegistry extends SynchronousRegistry {
    translate(key) {
        return key;
    }
}
globalRegistry.set('i18n', new FakeI18NRegistry());

// Data Source DataLoader
export const MockDataSourceDataLoader = {
    _currentPromise: null,
    _currentPromiseResolveFn: null,

    // API in tests
    reset() {
        this._currentPromise = null;
    },

    resolveCurrentPromise(result) {
        this._currentPromiseResolveFn(result);
        return this._currentPromise;
    },

    // API of DataLoader
    resolveValue() {
        if (!this._currentPromise) {
            this._currentPromise = new Promise(resolve => {
                this._currentPromiseResolveFn = resolve;
            });
        }
        return this._currentPromise;
    }
};
globalRegistry.set('dataLoaders', new SynchronousRegistry());
globalRegistry.get('dataLoaders').set('DataSources', MockDataSourceDataLoader);

// wrapper
export class WrapWithMockGlobalRegistry extends PureComponent {
    static propTypes = {
        children: PropTypes.element.isRequired
    };

    static childContextTypes = {
        globalRegistry: PropTypes.object.isRequired,
        configuration: PropTypes.object.isRequired
    };

    getChildContext() {
        const configuration = {};
        return {globalRegistry, configuration};
    }

    render() {
        return Children.only(this.props.children);
    }
}
