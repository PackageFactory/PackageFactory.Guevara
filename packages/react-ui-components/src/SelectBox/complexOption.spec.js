import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {createStubComponent} from './../_lib/testUtils.js';
import ComplexOption from './complexOption.js';

describe('<ComplexOption/>', () => {
    let props;

    beforeEach(() => {
        props = {
            label: 'Foo label',
            option: {},
            onClick: jest.fn(),
            theme: {},
            IconComponent: createStubComponent()
        };
    });

    it('should render correctly.', () => {
        const wrapper = shallow(<ComplexOption {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
