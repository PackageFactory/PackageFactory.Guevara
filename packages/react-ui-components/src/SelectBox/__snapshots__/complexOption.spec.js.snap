// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComplexOption/> should render correctly. 1`] = `
<SelectBoxOption
  IconComponent={[Function]}
  className="complexOption__item"
  label="Foo label"
  onClick={[Function]}
  option={Object {}}
  theme={Object {}}
>
  <span>
    Foo label
  </span>
</SelectBoxOption>
`;
